inst/stan/functions/delay_lpmf.stan

```diff
@@ -27,33 +27,59 @@ real delay_group_lpmf(array[] int groups, int start, int end, array[] int obs,
                       array[] real phi, int model_obs, int model_miss,
-                      vector miss_ref_lprop) {
-  // For missing model
-  // Passing in missing prop
-  // Pass in missing obs
-  // Make model optional
-  // need to know which report date data is linked (i.e we need a new vector
-  // look-up) by group
-  // sum by this look-up to get cases by report date and group (can also make
-  // this a matrix)
-  // scale all data by missing proportion or 1 minus missing proportion
-
+                      int miss_obs, array[] int missing_reference,
+                      array[,] int obs_by_report, vector miss_ref_lprop,
+                      array[] int sdmax, array[] int csdmax) {
   // Where am I?
   real tar = 0;
   int i_start = ts[1, start];
   int i_end = ts[t, end];
   array[3] int n = filt_obs_indexes(i_start, i_end, csl, sl);
-  array[n[3]] int filt_obs = obs[(n[1] + 1):n[2]];
+  array[n[3]] int filt_obs = segment(obs, n[1], n[3]);
+  // What is going to be used for storage
   vector[n[3]] log_exp_obs;
+  vector[model_miss ? miss_obs : 0]  log_exp_obs_miss;
 
-  // Combine expected final obs and date effects to get expected obs
-  log_exp_obs = expected_obs_from_snaps(
-    i_start, i_end, imp_obs, rdlurd, srdlh, ref_lh, dpmfs, ref_p, rep_h, ref_as_p, sl, csl, sg, st, n[3]
-  );
+  // Combine expected final obs and time effects to get expected obs
+  // If missing reference module in place calculate all expected obs vs 
+  // just those observed and allocate if missing or not.
+  if (model_miss) {
+    array[3] int f = filt_obs_indexes(i_start, i_end, csdmax, sdmax);
+    vector[f[3]] log_exp_all;
+
+    // Calculate all expected observations
+    log_exp_all = expected_obs_from_snaps(
+      i_start, i_end, imp_obs, rdlurd, srdlh, ref_lh, dpmfs, ref_p, rep_h, ref_as_p, sdmax, csdmax, sg, st, f[3]
+    );
 
+    // Allocate to just those actually observed
+    log_exp_obs = allocate_observed_obs(
+      i_start, i_end, log_exp_all, sl, csl, sdmax, csdmax
+    );
+    log_exp_obs = apply_missing_reference_effects(
+      i_start, i_end, log_exp_obs, sl, csl, log1m_exp(miss_ref_lprop)
+    );
+
+    // Allocate expected cases by reporting time
+    if (miss_obs) {
+      log_exp_all = apply_missing_reference_effects(
+        i_start, i_end, log_exp_all, sdmax, csdmax, miss_ref_lprop
+      );
+      log_exp_obs_miss = log_expected_by_report(
+        log_exp_all, obs_by_report
+      );
+    }
+  }else{
+    log_exp_obs = expected_obs_from_snaps(
+      i_start, i_end, imp_obs, rdlurd, srdlh, ref_lh, dpmfs, ref_p, rep_h, ref_as_p, sl, csl, sg, st, n[3]
+    );
+  }
   // Observation error model (across all reference times and groups)
   profile("model_likelihood_neg_binomial") {
   tar = obs_lpmf(filt_obs | log_exp_obs, phi, model_obs);
+  if (model_miss && miss_obs) {
+    tar += obs_lpmf(missing_reference | log_exp_obs_miss, phi, model_obs);
+  }
```

Comment:
@seabbs Are we maybe still missing the filtering for the current group here? I.e. the parallel to 

```
array[3] int n = filt_obs_indexes(i_start, i_end, csl, sl);
array[n[3]] int filt_obs = segment(obs, n[1], n[3]);
```
for `missing_reference`?

Reply:
ah, yes I think you are right. I don't see how this can be correct as written when there are multiple groups present.

Reply:
Note this shouldn't be linked to the issues we see as all testing so far done on only a single group.

Reply:
Yeah, agree, this is a different topic

Reply:
I've resolved this in https://github.com/epinowcast/epinowcast/pull/147/commits/2ec8058155fa4b8f6ddc2f3833803b3d90ef93e9